File: lib/src/main/java/com/alfianyusufabdullah/spager/widget/PageAdapter.java
package com.alfianyusufabdullah.spager.widget;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Created by jonesrandom on 12/19/17.
 * <p>
 * AA
 */

class PageAdapter extends FragmentPagerAdapter {

    private List<PageModel> pages;

    public PageAdapter(FragmentManager fm) {
        super(fm);
    }

    public void setPages(List<PageModel> pages) {
        this.pages = pages;
    }

    @Override
    public Fragment getItem(int position) {
        return pages.get(position).getPage();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return pages.get(position).getTitle();
    }

    @Override
    public int getCount() {
        return pages.size();
    }
}

File: lib/src/main/java/com/alfianyusufabdullah/PageAdapter.kt
package com.alfianyusufabdullah

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Created by jonesrandom on 3/24/18.
 *
 * @site www.androidexample.web.id
 * @github @alfianyusufabdullah
 */

internal class PageAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private var pages: MutableList<PageModel> = mutableListOf()

    fun setPages(pages: MutableList<PageModel>) {
        this.pages = pages
    }

    override fun getItem(position: Int): Fragment {
        return pages[position].page
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return pages[position].title
    }

    override fun getCount(): Int {
        return pages.size
    }
}
